crates/cubecl-core/src/frontend/element/cast.rs

```diff
@@ -29,3 +32,32 @@ impl<P: CubePrimitive> Cast for P {
         unexpanded!()
     }
 }
+
+pub trait BitCast: CubePrimitive {
+    #[allow(unused_variables)]
+    fn bitcast_from<From: CubePrimitive>(value: From) -> Self {
+        unexpanded!()
+    }
+
+    fn __expand_bitcast_from<From>(
+        context: &mut CubeContext,
+        value: From,
```

Comment:
Theorically, it should be `From::ExpandType`.

Reply:
I kept it the same as the `Cast` type. It also uses the plain generic type.